tests/Unit/Core/Form/IdentifiableObject/DataProvider/CombinationFormDataProviderTest.php

```diff
@@ -0,0 +1,251 @@
+    private const COMBINATION_ID = 42;
+    private const DEFAULT_QUANTITY = 51;
+
+    public function testGetDefaultData()
+    {
+        $queryBusMock = $this->createMock(CommandBusInterface::class);
+        $provider = new CombinationFormDataProvider($queryBusMock);
+
+        $this->assertEquals([], $provider->getDefaultData());
+    }
+
+    /**
+     * @dataProvider getExpectedData
+     *
+     * @param array $combinationData
+     * @param array $expectedData
+     */
+    public function testGetData(array $combinationData, array $expectedData)
+    {
+        $queryBusMock = $this->createQueryBusMock($combinationData);
+        $provider = new CombinationFormDataProvider($queryBusMock);
+
+        $formData = $provider->getData(static::COMBINATION_ID);
+        // assertSame is very important here We can't assume null and 0 are the same thing
+        $this->assertSame($expectedData, $formData);
+    }
+
+    public function getExpectedData(): Generator
+    {
+        $datasetsByType = [
+            $this->getDatasetsForStock(),
+        ];
+
+        foreach ($datasetsByType as $datasetByType) {
+            foreach ($datasetByType as $dataset) {
+                yield $dataset;
+            }
+        }
+    }
+
+    /**
+     * @return array
+     */
+    private function getDatasetsForStock(): array
+    {
+        $datasets = [];
+
+        $expectedOutputData = $this->getDefaultOutputData();
+        $combinationData = [];
+
+        $datasets[] = [
+            $combinationData,
+            $expectedOutputData,
+        ];
+
+        $expectedOutputData = $this->getDefaultOutputData();
+        $combinationData = [
+            'quantity' => 42,
+            'minimal_quantity' => 7,
+            'low_stock_threshold' => 5,
+            'low_stock_alert' => true,
+            'location' => 'top shelf',
+            'available_date' => new DateTime('1969/07/20'),
+        ];
+        $expectedOutputData['stock']['quantity'] = 42;
+        $expectedOutputData['stock']['minimal_quantity'] = 7;
+        $expectedOutputData['stock']['low_stock_threshold'] = 5;
+        $expectedOutputData['stock']['low_stock_alert'] = true;
+        $expectedOutputData['stock']['stock_location'] = 'top shelf';
+        $expectedOutputData['stock']['available_date'] = '1969-07-20';
+
+        $datasets[] = [
+            $combinationData,
+            $expectedOutputData,
+        ];
+
+        return $datasets;
+    }
+
+    /**
+     * @param array $combinationData
+     *
+     * @return MockObject|CommandBusInterface
+     */
+    private function createQueryBusMock(array $combinationData)
+    {
+        $queryBusMock = $this->createMock(CommandBusInterface::class);
+
+        $queryBusMock
+            ->method('handle')
+            ->with($this->logicalOr(
+                $this->isInstanceOf(GetCombinationForEditing::class)
+            ))
+            ->willReturnCallback(function ($query) use ($combinationData) {
```

Comment:
```suggestion
            ->willReturnCallback(function (GetCombinationForEditing $query) use ($combinationData) {
```

Reply:
Still same issue, multiple types are expected here, so I can't use strong typing because I'll have to remove them in the next PR (which doesn't make sense)